src/ipc/handler.c

```diff
@@ -677,6 +677,13 @@ static int ipc_pm_gate(uint32_t header)
 	else
 		pm_runtime_enable(PM_RUNTIME_DSP, PLATFORM_MASTER_CORE_ID);
 
+#if !CONFIG_CAVS_LPRO
+	if (pm_runtime_is_active(PM_RUNTIME_DSP, PLATFORM_MASTER_CORE_ID))
```

Comment:
If there is a risk of having unbalanced pm_gate IPC requests, pm_runtime_disable/enable calls should be protected with client bitmap, not a simple ref-counter (see how the put/get on the DSP resource is protected).